kythe/go/platform/tools/kzip_validator/kzip_validator.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Binary kzip_validator checks the contents of a kzip against a code repo and
+// compares file coverage.
+//
+// Example:
+//  kzip_validator -kzip <kzip-file> -local_repo <repo-root-dir> -lang cc,h
+//  kzip_validator -kzip <kzip-file> -repo_url <repo-url> -lang java [-version <hash>]
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"kythe.io/kythe/go/extractors/validation"
+)
+
+var (
+	kzip      = flag.String("kzip", "", "The kzip file to check")
+	repoURL   = flag.String("repo_url", "", "The repo to compare against")
+	version   = flag.String("version", "", "The version of the remote repo to compare")
+	localRepo = flag.String("local_repo", "", "The path of an optional local repo to specify instead of -repoURL")
+	lang      = flag.String("lang", "", "The comma-separated language file extensions to check, e.g. 'java' or 'cpp,h'")
+
+	missingFile = flag.String("missing_file", "", "An optional file to write all missing filepaths to")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %s -kzip <kzip-file> -repoURL <url> [-version <hash>]
+
+Compare a kzip file's contents with a given repo, and print results of file coverage.
```

Comment:
Please mention that the repo will be fetched (to where?) if necessary. That means we're invoking git which means we're sensitive to ambient config like `~/.gitconfig`, `~/.ssh/config`, etc. (You don't need to mention all of that, but it's helpful to know that the tool is calling `git`).

Reply:
This is in a future change.  And actually first version of that probably will just go for an archive zip (github) /tar.gz (internal mirror), not git commandline.

Right now the --repo_url version just fails

Reply:
Could we just leave that flag out for now, then?